apps/s_server.c

```diff
@@ -3004,20 +3005,26 @@ static void print_connection_info(SSL *con)
     (void)BIO_flush(bio_s_out);
 }
 
-#ifndef OPENSSL_NO_DH
-static DH *load_dh_param(const char *dhfile)
+static EVP_PKEY *load_dh_param(const char *dhfile)
 {
-    DH *ret = NULL;
+    EVP_PKEY *dhpkey = NULL;
     BIO *bio;
+    OSSL_DECODER_CTX *decoderctx = NULL;
 
     if ((bio = BIO_new_file(dhfile, "r")) == NULL)
         goto err;
-    ret = PEM_read_bio_DHparams(bio, NULL, NULL, NULL);
- err:
+
+    decoderctx
+        = OSSL_DECODER_CTX_new_by_EVP_PKEY(&dhpkey, "PEM", "DH",
+                                           NULL, NULL);
+    if (decoderctx != NULL)
+        OSSL_DECODER_from_bio(decoderctx, bio);
```

Comment:
There is now a `load_keyparams` in libapps...  does it still make sense to have a specialized `load_dh_params`?

Reply:
I've made `load_dh_params` a simple wrapper around `load_keyparams` which additionally confirms that we have DH parameters.